cachito/workers/pkg_managers/gomod.py

```diff
@@ -187,7 +189,47 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
             )
             shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
 
-        return module, deps
+        # package level dependencies
+        go_list_deps_output = run_gomod_cmd(
+            ("go", "list", "-deps", "-f", "{{if not .Standard}}{{.ImportPath}} {{.Module}}{{end}}"),
+            run_params,
+        )
+
+        pkg_level_deps = []
+        for line in go_list_deps_output.splitlines():
+            # line example: pkg.io/foo/bar pkg.io/foo/var v1.0.0 => pkg.io/foo/bar v1.0.1
+            parts = [part for part in line.split(" ") if part not in ("")]
+            replaces = None
+
+            name = parts[0]
+            parent_module = parts[1]
+            if len(parts) > 2:
+                version = parts[2]
+            else:
+                # use requested module version
+                version = module_version
+
+            if len(parts) > 3:
+                # parts[3] is just an arrow "=>" pointing to the replacements
+                replaces = {"type": "gomod", "name": parent_module, "version": version}
+                parent_module = parts[4]
+                version = parts[5]
+
+            pkg = {
+                "name": name,
+                "from_module": parent_module,
+                "replaces": replaces,
+                "type": "go-package",
+                "version": version,
+            }
+
+            pkg_level_deps.append(pkg)
+
+        # The last item on `go list -deps` is the main package being evaluated
+        package = pkg_level_deps.pop()
```

Comment:
Could there be more than one package in the module? How is this accounted for? Does it need to be?

Reply:
Yes. The sub-packages would be accounted for in the package level dependencies as the other (external) package level dependenies.